sechub-pds-examples/gosec/docker/Dockerfile

```diff
@@ -0,0 +1,94 @@
+ARG WORKSPACE="/workspace"
+ARG PDS_FOLDER="/pds"
+ARG TOOL_FOLDER="/tool"
+ARG PDS_VERSION=0.21.0
+ARG GOSEC_VERSION=2.8.0
+ARG GO=go1.16.5.linux-amd64.tar.gz
+
+# non-root user
+# using fixed group and user ids
+# gosec needs a home directory for the cache
+RUN groupadd --gid 2323 gosec \
+     && useradd --uid 2323 --no-log-init --create-home --gid gosec gosec
+
+# Update image and install dependencies
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
+    apt-get dist-upgrade --assume-yes && \
+    apt-get install --assume-yes wget openjdk-11-jre-headless && \
+    apt-get clean
+
+# Create tool folder
+RUN mkdir -p $TOOL_FOLDER
+COPY gosec.sh $TOOL_FOLDER/gosec.sh
+RUN chmod +x $TOOL_FOLDER/gosec.sh
+
+# Install Go
+RUN cd /tmp && \
+    # create checksum file
+    echo "b12c23023b68de22f74c0524f10b753e7b08b1504cb7e417eccebdd3fae49061 $GO" > $GO.sha256sum && \
+    # download Go
+    wget https://dl.google.com/go/${GO} && \
+    # verify that the checksum and the checksum of the file are same
+    sha256sum --check $GO.sha256sum && \
+    # extract Go
+    tar --extract --file $GO --directory "/usr/local/" && \
+    # remove go tar.gz
+    rm $GO && \
+    # add Go to path
+    echo 'export PATH="/usr/local/go/bin:$PATH":' >> /root/.bashrc
+
+# Install GoSec
+RUN cd /tmp && \
+    # download gosec
+    wget https://github.com/securego/gosec/releases/download/v${GOSEC_VERSION}/gosec_${GOSEC_VERSION}_linux_amd64.tar.gz && \
+    # download checksum
+    wget https://github.com/securego/gosec/releases/download/v${GOSEC_VERSION}/gosec_${GOSEC_VERSION}_checksums.txt && \
+    # verify checksum
+    sha256sum --check --ignore-missing "gosec_${GOSEC_VERSION}_checksums.txt" && \
+    # create gosec folder
+    mkdir -p "$TOOL_FOLDER/gosec" && \
+    # unpack gosec
+    tar --extract --ungzip --file "gosec_${GOSEC_VERSION}_linux_amd64.tar.gz" --directory "$TOOL_FOLDER/gosec" && \
+    # Remove gosec tar.gz
+    rm "gosec_${GOSEC_VERSION}_linux_amd64.tar.gz"
+    
+
+# Install the Product Delegation Server (PDS)
+RUN mkdir -p pds && \
+    cd /pds && \
+    # create checksum file
+    echo "f3101a6a2a96b39edf493249f2300c627fa01fcb7a6b96c5ae1edfbe70a31d46  sechub-pds-$PDS_VERSION.jar" > sechub-pds-$PDS_VERSION.jar.sha256sum && \
+    # download pds
+    wget "https://github.com/Daimler/sechub/releases/download/v$PDS_VERSION-pds/sechub-pds-$PDS_VERSION.jar" && \
+    # verify that the checksum and the checksum of the file are same
+    sha256sum --check sechub-pds-$PDS_VERSION.jar.sha256sum
+
+# Create the PDS workspace
+RUN mkdir -p $WORKSPACE    
+
+# Copy PDS configfile
+COPY pds-config.json /$PDS_FOLDER/pds-config.json
+
+# Copy run script into container
+COPY run.sh /run.sh
+RUN chmod +x /run.sh
+
+# Change owner of tool, workspace and pds folder as well as /run.sh
+RUN chown --recursive gosec:gosec $TOOL_FOLDER $WORKSPACE $PDS_FOLDER /run.sh
+
+WORKDIR $WORKSPACE
+
+# switch from root to non-root user
+USER gosec
+
+CMD ["/run.sh"]
```

Comment:
You could pass `$PDS_VERSION` to this script. This way you don't have to change this version at two separate places.

Reply:
did set the `$PDS_VERSION` as `ENV` variable in the script, which has the desired effect.